### ISSUE TITLE
Accepting CLAP_EVENT_TRANSPORT events or not has side effects elsewhere

### ISSUE THREAD

**Issue by robbert-vdh** [free-audio/clap#85]
According to `ext/event-filter.h`, this is the only event where accepting it has side effects:

https://github.com/free-audio/clap/blob/a80522ef56c2cdaab9f9dfc8dcf206ceb75d60a1/include/clap/ext/event-filter.h#L18

If you don't accept events for inter-buffer transport changes (which not all plugins might want to support), then you risk not getting any transport information at all. The alternative would be to falsely advertise that you support and to just ignore the events, but that feels very wrong.

**Comment by robbert-vdh** [comment]
Not accepting the event would simply cause the host to split the buffer, just like it would already do for VST2 and VST3 plugins. I think side effects like this is a thing CLAP should try to avoid at (almost) all costs. Having a lot of implicit behavior is what ends up creating plugin and host implementation bugs.

**Comment by Bremmers** [comment]
I think plugins should use the events. They tell you exactly when something happened. This wasn't possible until now (with other formats), but now it is, and it's better.

If you'd want to optimize things there should be a way to say you want the loop points, but you don't want the tempo (for example). Don't know if that's worth adding.

**Comment by robbert-vdh** [comment]
Yes, maybe plugins should, but right now only a handful of plugins use mid-buffer events. Until JUCE and the other big plugin frameworks support this (also also internal plugin frameworks), you can't assume that these events will actually be handled. The JUCE CLAP extension for instance will currently just ignore any `CLAP_EVENT_TRANSPORT` events you send it.

And my point was not that having `CLAP_EVENT_TRANSPORT` events is a bad idea. It's a great idea. My point was that side effects like this are a design smell. Accepting an event or not should not be able to cause a field in a struct that isn't directly involved with the event handling to become a null pointer.

**Comment by Bremmers** [comment]
Yeah I do get that it can be a bit unexpected that clap_process.transport disappears if you opt out of CLAP_EVENT_TRANSPORT. But it's documented ;-)

The reason it works this way is optimization: it allows the host to skip the transport calculations. If you'd change it there should ideally be another way to tell the host clap_process.transport isn't required. Or more specifically, that the time consuming tempo/timesig bits of clap_process.transport aren't required.

**Comment by robbert-vdh** [comment]
Both VST2 and VST3 let the plugin tell the host exactly what transport information it's interested in (and presumably other plugin APIs have a similar feature).

**Comment by Bremmers** [comment]
Perhaps event-filter should be changed to something that can filter other things too (things that are not events) too. With a bunch of constants. Then you could opt out of certain clap_event_transport fields, or opt out of clap_process.transport.

 It could have a name that better describes it's purpose, like 'optimization'.

**Comment by Bremmers** [comment]
How about this:

- rename event-filter.h to optimizations.h
- clap_plugin_event_filter.accepts() now becomes clap_plugin_optimizations.accepts_event()
- clap_plugin_optimizations.optimizations function is added, returns UINT32. Flags can be defined for this:

CLAP_OPTIMIZATION_NO_CLAP_PROCESS_TRANSPORT = 1; //don't set clap_process.transport
CLAP_OPTIMIZATION_NO_CLAP_EVENT_TRANSPORT_TEMPO_TIMESIG = 2;//don't fill clap_event_transport.tempo, tempo_inc, tsig_num, tsig_denom

**Comment by abique** [comment]
Yo, I moved event-filter back to draft.
I think that we don't need event-filter at all.

**Comment by abique** [comment]
Regarding the transport events, I think it is not a problem because the plugin can break the process call itself:

```
process(int nframes) {
  for (int i = 0; i < transportEventTime;) {
     int tranportEventTime = findTransportEvent(i, nframes); // returns nframes if not found
     doProcess(i, transportEventTime - i);
     i = transportEventTime;
  }
}

doProcess(int offset, int nframes) {
  /* processes nframes, no transport in the middle change here */
}
```

**Comment by abique** [comment]
I actually removed the event-filter extension.

**Comment by robbert-vdh** [comment]
Then this can be closed!
